monai/inferers/inferer.py

```diff
@@ -221,3 +221,68 @@ def __call__(self, inputs: torch.Tensor, network: nn.Module, *args: Any, **kwarg
             cam = GradCAMpp(network, self.target_layers, *self.args, **self.kwargs)
 
         return cam(inputs, self.class_idx, *args, **kwargs)
+
+
+class SliceInferer(SlidingWindowInferer):
+    """
+    SliceInferer extends SlidingWindowInferer to provide slice-by-slice (2D) inference
+    when provided a 3D volume.
+
+    Args:
+        spatial_dim: Spatial dimension over which the slice-by-slice inference runs on the 3D volume.
+            For example ``0`` could slide over axial slices. ``1`` over coronal slices and ``2`` over sagittal slices.
+        args: other optional args to be passed to the `__init__` of SliceInferer.
+        kwargs: other optional keyword args to be passed to `__init__` of SliceInferer.
+
+
+    ``args``, ``kwargs`` follow :py:class:`monai.inferer.SlidingWindowInferer`.
+
+    """
+
+    def __init__(self, spatial_dim: int = 0, *args, **kwargs) -> None:
+        self.spatial_dim = spatial_dim
+        super().__init__(*args, **kwargs)
+
+    def __call__(
+        self, inputs: torch.Tensor, network: Callable[..., torch.Tensor], *args: Any, **kwargs: Any
+    ) -> torch.Tensor:
+        """
+        Args:
+            inputs: 3D input for inference
+            network: 2D model to execute inference on slices in the 3D input
+            args: optional args to be passed to ``network``.
+            kwargs: optional keyword args to be passed to ``network``.
+        """
+        assert self.spatial_dim < 3, "`spatial_dim` can only be `[D, H, W]` with `0, 1, 2` respectively"
+
+        # Check if roi size (eg. 2D roi) and input volume sizes (3D input) mismatch
+        if len(self.roi_size) != len(inputs.shape[2:]):
```

Comment:
Could you please help add some doc-string to describe the expected `roi_size`?

Thanks.